### ISSUE TITLE
IDE-PHP language server stopped unexpectedly. Without any description (Blank)

### ISSUE THREAD

**Issue by wakamin** [atom/ide-php#91]
Hi,

I am using:
- Atom 1.23.1
- ide-php 0.7.4
- PHP 7.1.12

Setting:
- Socket: auto
- Memory limit: -1

IDE PHP us throwing error "IDE-PHP language server stopped unexpectedly" without any description, it is just blank. It is also happen in PHP 7.2.

**Comment by rsese** [comment]
Thanks for the report - do you have any specific steps that reproduce the error?  Also, can you share what operating system you're running and check if there any more errors in developer tools console [1]?

[1]: http://flight-manual.atom.io/hacking-atom/sections/debugging/#check-for-errors-in-the-developer-tools

**Comment by wakamin** [comment]
Steps to reproduce error:
- I open file of my Laravel Project, then php-ide is loading says "Installing... FelixFBecker initializing my-project"
- Then the error occurs.

OS:
- Elementary OS with php 7.1.12.
- It is also have the same error in my Arch Linux with php 7.2

Error in Dev tools:
GET https://github.com/atom/ide-php/files/1549031/php-ls-5.2.0.tar.gz net::ERR_TIMED_OUT
downloadFile @ download-file.js:26
fileExists.then.then @ /home/user/.atom/packages/ide-php/lib/main.js:137

**Comment by wakamin** [comment]
Helo,

Any update on this issue? I can download the file manually using browser, but Atom cannot download it.

**Comment by damieng** [comment]
No idea what could be causing it to just time out there when it works fine in your browser.  :(

**Comment by picasso250** [comment]
Same problem

- Atom 1.23.1
- Win10 1607
- php-7.0.26-Win32-VC14-x64

I set PHP path to 
`D:/Software/php-7.0.26-Win32-VC14-x64/php.exe`

**Comment by wakamin** [comment]
In my case it seems the problem is on my ISP. It is working when I change internet connection to another ISP. For me it was solved.

**Comment by picasso250** [comment]
After hours of work, I find that it is because my php.exe is never exuceted as a server, like
`php -S 0.0.0.0:80`
When first time it is executed, Windows asks firewall settings for it, which by defaut, is blocked.
So I executed as buildin server and allow it, the problem seems solved.

**Comment by damieng** [comment]
Closing this issue.  Should consider adding this to a troubleshooting section.

**Comment by softwarecreations** [comment]
I also had this issue now, while I was offline. I restarted atom but the issue kept occurring.
After I connected to the internet I restarted atom, and then the issue did not occur.
I restarted numerous times online and offline and the issue did not re-occur so unfortunately I can't debug it.

Does it perhaps try get something from the internet? Is that the cause of the bug?

I tried to use @picasso250 's suggestion of `/usr/bin/php -S 0.0.0.0:1234` (port 80 is taken) but it didn't like that and is only happy with `/usr/bin/php`.

I'm not sure if the issue occurred with `atom-autocomplete-php 2.25.6` or `ide-php 0.7.14`
